src/app/components/normal-component/demo-page/demo-card.component.ts

```diff
@@ -0,0 +1,44 @@
+import { AfterViewInit, Component, ComponentFactory, ComponentFactoryResolver, ComponentRef, Input, Type, ViewChild, ViewContainerRef } from '@angular/core';
+import { AbstractNode } from 'src/app/jscaip/MGPNode';
+import { Utils } from 'src/app/utils/utils';
+import { AbstractGameComponent } from '../../game-components/game-component/GameComponent';
+
+export type DemoNodeInfo = {
+    name: string, // The name of the game
+    component: Type<AbstractGameComponent>, // The component
+    node: AbstractNode, // The demo node
+}
+
+@Component({
+    selector: 'app-demo-card',
+    template: `<div #board></div>`,
+})
+export class DemoCardComponent implements AfterViewInit {
+    @Input() public demoNodeInfo: DemoNodeInfo;
+
+    @ViewChild('board', { read: ViewContainerRef })
+    public boardRef: ViewContainerRef | null;
+
+    public gameComponent: AbstractGameComponent | null;
+
+    public constructor(private readonly componentFactoryResolver: ComponentFactoryResolver) {
+    }
```

Comment:
Isn't that the Fourth Game Wrapper then ?

Reply:
Oui ! C'en est un maintenant (de GameWrapper)